src/object_storage/api.rs

```diff
@@ -55,11 +62,17 @@ where
     let c_id = container.as_ref();
     let o_id = object.as_ref();
     debug!("Creating object {} in container {}", o_id, c_id);
-    let _ = session.send_checked(
-        session
-            .request(OBJECT_STORAGE, Method::PUT, &[c_id, o_id], None)?
-            .body(SyncBody::new(body)),
-    )?;
+    let mut req = session.request(OBJECT_STORAGE, Method::PUT, &[&c_id, &o_id], None)?;
+
+    if headers.delete_after.is_some() {
```

Comment:
Rather than `is_some` followed by `unwrap`, the usual pattern is

```rust
if let Some(delete_after) = headers.delete_after {
  // do something with delete_after
}
```